src/interface/update.cpp

```diff
@@ -119,14 +127,118 @@ TaskStatus UpdateWithFluxDivergence(MeshData<Real> *u0_data, MeshData<Real> *u1_
       DEFAULT_LOOP_PATTERN, "UpdateWithFluxDivergenceMesh", DevExecSpace(), 0,
       u0_pack.GetDim(5) - 1, 0, u0_pack.GetDim(4) - 1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
       KOKKOS_LAMBDA(const int m, const int l, const int k, const int j, const int i) {
-        const auto &coords = u0_pack.coords(m);
-        const auto &u0 = u0_pack(m);
-        u0_pack(m, l, k, j, i) = gam0 * u0(l, k, j, i) + gam1 * u1_pack(m, l, k, j, i) +
-                                 beta_dt * FluxDivHelper(l, k, j, i, ndim, coords, u0);
+        if (u0_pack.IsAllocated(m, l) && u1_pack.IsAllocated(m, l)) {
+          const auto &coords = u0_pack.coords(m);
+          const auto &u0 = u0_pack(m);
+          u0_pack(m, l, k, j, i) = gam0 * u0(l, k, j, i) + gam1 * u1_pack(m, l, k, j, i) +
+                                   beta_dt * FluxDivHelper(l, k, j, i, ndim, coords, u0);
+        }
       });
   return TaskStatus::complete;
 }
 
+TaskStatus SparseDeallocCheck(MeshData<Real> *md) {
+  if (!Globals::sparse_config.enabled || (md->NumBlocks() == 0)) {
+    return TaskStatus::complete;
+  }
+
+  Kokkos::Profiling::pushRegion("Task_SparseDeallocCheck");
+
+  const IndexRange ib = md->GetBoundsI(IndexDomain::entire);
+  const IndexRange jb = md->GetBoundsJ(IndexDomain::entire);
+  const IndexRange kb = md->GetBoundsK(IndexDomain::entire);
```

Comment:
```suggestion
  const IndexRange ib = md->GetBoundsI(IndexDomain::interior);
  const IndexRange jb = md->GetBoundsJ(IndexDomain::interior);
  const IndexRange kb = md->GetBoundsK(IndexDomain::interior);
```

Given that we also use the threshold for sending, I think we restrict the deallocation to non-ghost zones.

Reply:
Hmm, making this change breaks the restart and sparse tests (sparse results don't agree with dense results up to allocation tolerance). So I'll keep this as `entire`. We might try to understand this better in the future.